Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -295,16 +423,30 @@ public struct HTTPHeaders: CustomStringConvertible {
     /// - Parameter name: The header field name whose values are to be retrieved.
     /// - Returns: A list of the values for that header field name.
     public func getCanonicalForm(_ name: String) -> [String] {
+        let result = self[name]
+
         // It's not safe to split Set-Cookie on comma.
-        let queryName = name.lowercased()
-        if queryName == "set-cookie" {
-            return self[name]
+        guard !name.equalCaseInsensitiveASCII("set-cookie") else {
+            return result
+        }
+
+        return result.map { $0.split(separator: ",").map { String($0.trimWhitespace()) } }.reduce([], +)
```

Comment:
This should be rebased onto master, which contains an improvement in this code to use `flatMap` instead of `map.reduce`.